src/order/ideal.lean

```diff
@@ -96,6 +106,22 @@ lemma proper_of_not_mem {I : ideal P} {p : P} (nmem : p ∉ I) : proper I :=
   exact nmem (set.mem_univ p),
 end⟩
 
+/-- A maximal ideal if it is maximal in the collection of proper ideals.
+  Note that we cannot use the `is_coatom` class because `P` might not have a `top` element.
+-/
+def is_maximal (I : ideal P) : Prop :=
+  proper I ∧ ∀ J : ideal P, I < J → J.carrier = ⊤
+
+lemma is_maximal_iff {I : ideal P} :
+  is_maximal I ↔ proper I ∧ ∀ J : ideal P, I < J → J.carrier = ⊤ :=
+⟨λ h, ⟨h.1, h.2⟩, λ h, ⟨h.1, h.2⟩⟩
+
+lemma is_maximal.proper {I : ideal P} (hI : is_maximal I) : proper I := hI.1
+
+lemma is_maximal.maximal_proper {I : ideal P} (hI : is_maximal I) :
+  ∀ J : ideal P, I < J → J.carrier = ⊤
+:= hI.2
+
```

Comment:
Using a structure and the `mk_iff` attribute covers all of the declarations below. Also, using `set.univ` might be less confusing than using `⊤` here.
```suggestion
@[mk_iff] structure is_maximal (I : ideal P) : Prop :=
(proper : proper I)
(maximal_proper : ∀ J : ideal P, I < J → J.carrier = set.univ)

```